src/main/java/Naega.java

```diff
@@ -0,0 +1,150 @@
+
+        scanner.close();
+    }
+
+    private static void handleList(ArrayList<Task> tasks) {
+        System.out.println("____________________________________________________________");
+        System.out.println(" Here are the tasks in your list:");
+        for (int i = 0; i < tasks.size(); i++) {
+            System.out.println(" " + (i + 1) + "." + tasks.get(i));
+        }
+        System.out.println("____________________________________________________________");
+    }
+
+    private static void handleMark(ArrayList<Task> tasks, String userInput) throws NaegaException {
+        int taskIndex = parseTaskIndex(userInput, "mark ");
+        tasks.get(taskIndex).markAsDone();
+        System.out.println("____________________________________________________________");
+        System.out.println(" Nice! I've marked this task as done:");
+        System.out.println("   " + tasks.get(taskIndex));
+        System.out.println("____________________________________________________________");
+    }
+
+    private static void handleUnmark(ArrayList<Task> tasks, String userInput) throws NaegaException {
+        int taskIndex = parseTaskIndex(userInput, "unmark ");
+        tasks.get(taskIndex).markAsNotDone();
+        System.out.println("____________________________________________________________");
+        System.out.println(" OK, I've marked this task as not done yet:");
+        System.out.println("   " + tasks.get(taskIndex));
+        System.out.println("____________________________________________________________");
+    }
+
+    private static void handleTodo(ArrayList<Task> tasks, String userInput) throws NaegaException {
+        String description = parseDescription(userInput, "todo ");
+        Task newTask = new Todo(description);
+        tasks.add(newTask);
+        printTaskAdded(tasks, newTask);
+    }
+
+    private static void handleDeadline(ArrayList<Task> tasks, String userInput) throws NaegaException {
+        String[] parts = parseDescription(userInput, "deadline ").split(" /by ");
+        if (parts.length < 2) {
+            throw new NaegaException("The deadline must include a '/by' keyword.");
+        }
+        Task newTask = new Deadline(parts[0], parts[1]);
+        tasks.add(newTask);
+        printTaskAdded(tasks, newTask);
+    }
+
+    private static void handleEvent(ArrayList<Task> tasks, String userInput) throws NaegaException {
+        String[] parts = parseDescription(userInput, "event ").split(" /from ");
+        if (parts.length < 2) {
+            throw new NaegaException("The event must include a '/from' keyword.");
+        }
+        String description = parts[0];
+        String[] timeParts = parts[1].split(" /to ");
+        if (timeParts.length < 2) {
+            throw new NaegaException("The event must include a '/to' keyword.");
+        }
+        Task newTask = new Event(description, timeParts[0], timeParts[1]);
+        tasks.add(newTask);
+        printTaskAdded(tasks, newTask);
+    }
```

Comment:
Similar to the previous comment, perhaps, the common code could be extracted into a separate method?